### PR TITLE
Add optional client options to Anthropic sdk to allow overridable default headers

### PR DESCRIPTION
**Pull request by colegottdank**
Fixes # (issue)

### CODE REVIEW

File: langchain/src/chat_models/anthropic.ts [hunk 1]
```diff
@@ -159,7 +164,14 @@ export class ChatAnthropic extends BaseChatModel implements AnthropicInput {
   // Used for streaming requests
   private streamingClient: Anthropic;
 
-  constructor(fields?: Partial<AnthropicInput> & BaseChatModelParams) {
+  private clientOptions: ClientOptions;
+
+  constructor(
+    fields?: Partial<AnthropicInput> &
+      BaseChatModelParams & {
+        clientOptions?: ClientOptions;
```
Comment (jacoblee93) [inline langchain/src/chat_models/anthropic.ts:172]: Could we instead add `clientOptions?` to `AnthropicInput` directly?

**Comment by jacoblee93** [comment]
Seems good, just one question!

File: langchain/src/chat_models/anthropic.ts (hunk 1 above)
Comment (colegottdank) [reply langchain/src/chat_models/anthropic.ts]: Done & added test

**Comment by colegottdank** [comment]
> Seems good, just one question!

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**